### PR TITLE
Link fixes, log fix and yaml updates

### LINKED ISSUE #1418: reconciler: incomplete log output
this line 
https://github.com/intel/intel-device-plugins-for-kubernetes/blob/d60419e4f0063023ea3953c25336c0b2ee9eb9f8/pkg/controllers/reconciler.go#L233

is causing incomplete log lines that look suspicious especially when the diff is empty:
```
I0515 14:13:19.142299 1922029 reconciler.go:233] "" controller="dsadeviceplugin" controllerGroup="deviceplugin.intel.com" controllerKind="DsaDevicePlugin" DsaDevicePlugin="dsadeviceplugin-test" namespace="" name=
"dsadeviceplugin-test" reconcileID=d9fa9595-5651-4796-9c6d-75073106860e ="(MISSING)"
```

### LINKED ISSUE #1446: deployment with kustomize: fix deprecated bases, patchesStrategicMerge, vars
When deploying a plugin or an operator, some warning messages below apper:
>  \# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
  \# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
  \# Warning: 'vars' is deprecated. Please use 'replacements' instead. [EXPERIMENTAL] Run 'kustomize edit fix' to update your 
Kustomization automatically.

### LINKED ISSUE #1449: qat linux kernel link is broken
Link [here](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/qat_plugin/README.md) to the [Linux Kernel upstream drivers](https://github.com/torvalds/linux/tree/master/drivers/crypto/qat) hits 404.

### PR DESCRIPTION
**Pull request by tkatila**
This started as a PR where I wanted to push some dangling fixes, but the yaml updates turned out to be more extensive than I anticipated.

Fixes #1446 
Fixes #1449
Touches #1418

### CODE REVIEW

**Comment by mythi** [comment]
> Fixes https://github.com/intel/intel-device-plugins-for-kubernetes/issues/1418

agree but we probably want a new issue to rework the part that checks when an update is needed

File: test/e2e/utils/utils.go [hunk 1]
```diff
@@ -118,17 +119,83 @@ func LocateRepoFile(repopath string) (string, error) {
...
+		return err
+	}
+
+	// Copy all files under the kustomize path under the temp overlay path.
+	err = copyFiles(kustomizeYamlFileDir, overlayDir)
+	if err != nil {
+		return err
+	}
+
+	bytes, err := os.ReadFile(overlayDir + "/kustomization.yaml")
```
Comment (mythi) [inline test/e2e/utils/utils.go:159]: `filepath.Join()`

File: test/e2e/utils/utils.go [hunk 2]
```diff
@@ -118,17 +119,83 @@ func LocateRepoFile(repopath string) (string, error) {
...
+	}
+
+	content["resources"] = resources
+
+	bytes, err = yaml.Marshal(content)
+	if err != nil {
+		return err
+	}
+
+	if err := os.WriteFile(overlayDir+"/kustomization.yaml", bytes, 0600); err != nil {
```
Comment (mythi) [inline test/e2e/utils/utils.go:194]: `filepath.Join()`

File: test/e2e/utils/utils.go [hunk 3]
```diff
@@ -118,17 +119,83 @@ func LocateRepoFile(repopath string) (string, error) {
...
+			return err
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+// CreateKustomizationOverlay copies the base overlay, and changes the namespace
```
Comment (mythi) [inline test/e2e/utils/utils.go:145]: perhaps add a comment that any cleanup after copying is left to the caller?

**Review by mythi — COMMENTED** [review]
some minor suggestions.

File: test/e2e/utils/utils.go (hunk 3 above)
Comment (tkatila) [reply test/e2e/utils/utils.go:145]: Added

File: test/e2e/utils/utils.go [hunk 4]
```diff
@@ -118,17 +119,85 @@ func LocateRepoFile(repopath string) (string, error) {
...
+		return err
+	}
+
+	// Copy all files under the kustomize path under the temp overlay path.
+	err = copyFiles(kustomizeYamlFileDir, overlayDir)
+	if err != nil {
+		return err
+	}
+
+	kustomizationFile := filepath.Join(overlayDir, "/kustomization.yaml")
```
Comment (mythi) [inline test/e2e/utils/utils.go:159]:
```suggestion
	kustomizationFile := filepath.Join(overlayDir, "kustomization.yaml")
```

**Review by mythi — APPROVED** [review]
LGTM

**Review by mythi — APPROVED** [review]

**Review by hj-johannes-lee — APPROVED** [review]

**Pull request merged**